lib/castle/configuration.rb

```diff
@@ -58,6 +64,22 @@ def blacklisted=(value)
       @blacklisted = (value ? value.map { |header| @formatter.call(header) } : []).freeze
     end
 
+    # sets ip headers
+    # @param value [Array<String>]
+    def ip_headers=(value)
+      raise Castle::ConfigurationError, 'ip headers must be an Array' unless value.is_a?(Array)
+
+      @ip_headers = value.map { |header| @formatter.call(header) }.freeze
+    end
+
+    # sets trusted proxies
+    # @param value [Array<String|Regexp>]
+    def trusted_proxies=(value)
+      raise Castle::ConfigurationError, 'trusted proxies must be an Array' unless value.is_a?(Array)
+
```

Comment:
should we validate that the elements are Strings or Regexps too?

Reply:
array type check should be enough to cover possible misconfigurations